pkg/build/api/types.go

```diff
@@ -20,6 +20,30 @@ const (
 	DefaultDockerLabelNamespace = "io.openshift."
 )
 
+const (
+	// SecretTypeSource contains data needed to authenticate against SCM servers.
+	//
+	// SecretTypeSource keys are devided into these option groups:
+	//   Basic authentication group. Requires at least one of fields:
+	//     - Secret.Data["username"] - username used to authenticate against SCM server
+	//     - Secret.Data["password"] - password or token needed to authenticate against SCM server
+	//     - Secret.Data["ca.crt"] - CA certificate needed to authenticate against SCM server
+	//   SSH authentication group. Required field:
+	//     - Secret.Data["ssh-privatekey"] - private SSH key needed to authenticate against SCM server
+	SecretTypeSource kapi.SecretType = "Source"
```

Comment:
You can't because we have to run against external Kube.  Kube should have
more secret types and not reject types it doesn't know about.  Kube secret
types should be more flexible and allow unrelated secret data in them.

On Sep 9, 2015, at 8:47 AM, Cesar Wong <notifications@github.com> wrote:

In pkg/build/api/types.go
<https://github.com/openshift/origin/pull/4433#discussion_r39037198>:

> @@ -20,6 +20,30 @@ const (
>  	DefaultDockerLabelNamespace = "io.openshift."
>  )
>
> +const (
> +	// SecretTypeSource contains data needed to authenticate against SCM servers.
> +	//
> +	// SecretTypeSource keys are devided into these option groups:
> +	//   Basic authentication group. Requires at least one of fields:
> +	//     - Secret.Data["username"] - username used to authenticate against SCM server
> +	//     - Secret.Data["password"] - password or token needed to authenticate against SCM server
> +	//     - Secret.Data["ca.crt"] - CA certificate needed to authenticate against SCM server
> +	//   SSH authentication group. Required field:
> +	//     - Secret.Data["ssh-privatekey"] - private SSH key needed to authenticate against SCM server
> +	SecretTypeSource kapi.SecretType = "Source"

And if we define those types they should be defined upstream under the Kube
namespace

The thought was that we would allow expanding the upstream types via a
validator plugin.

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/4433/files#r39037198>.